pulp_ansible/app/tasks/collections.py

```diff
@@ -120,47 +122,24 @@ def import_collection(
     CollectionImport.objects.get_or_create(task_id=get_current_job().id)
 
     artifact = Artifact.objects.get(pk=artifact_pk)
-    log.info("Processing collection from {path}".format(path=artifact.file.path))
+    filename = CollectionFilename(expected_namespace, expected_name, expected_version)
+    log.info(f"Processing collection from {artifact.file.name}")
     import_logger = logging.getLogger("pulp_ansible.app.tasks.collection.import_collection")
 
-    try:
-        importer_result = process_collection(str(artifact.file.path), logger=import_logger)
-        collection_info = importer_result["metadata"]
-
-        if expected_namespace and expected_namespace != collection_info["namespace"]:
-            msg = _(
-                "Expected namespace '{expected_namespace}', but instead Collection has namespace "
-                "'{collection_namespace}'."
-            )
-            raise ValidationError(
-                msg.format(
-                    expected_namespace=expected_namespace,
-                    collection_namespace=collection_info["namespace"],
+    with _artifact_guard(artifact):
+        try:
+            with artifact.file.open() as artifact_file:
+                importer_result = process_collection(
+                    artifact_file, filename=filename, logger=import_logger
```

Comment:
How does the validation occur here? `expected_namespace`, `expected_name`, and `expected_version` don't seem to be used.

Reply:
the validation occurs within `galaxy-importer` and raises an error if there is no match: https://github.com/ansible/galaxy-importer/blob/master/galaxy_importer/collection.py#L103-L113

Reply:
That seems to use a filename check, which when used in pulp_ansible will not have the same filename structure as saved in the content addressable storage. Am I reading this right?

Reply:
Since expected namespace, name and version are passed into this method, the filename structure could be different before this method is called. Those 3 parameters are passed into galaxy-importer then each, if populated, are validated against metadata.

Reply:
I see now. The filename is formed from the `expected_namespace`, etc on L#125. So this should work for pulp_ansible just like before.